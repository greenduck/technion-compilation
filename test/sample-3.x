/*This is an example */

integer function (param5:real, param2:integer)
var b:integer; /*c:real; d:integer;*/

begin
LeftAssn = (call(0.44 , 15))*(plusParam1 + plusParam2)/(minus1 - minus2);
if a==b || c<d || e<>f then x=y;
for (i=0;i<size;i=i+1;)
	read (book);
while !(tierd==tierd) || ran<yoni do
		return (coffie); 
if first==exp then if second==exp2 then return (first); else write (letter);

end
program
var a:integer;
begin
	/*if I<Y then if a == b then A=B; else read(AB); else write(IY);*/

end



