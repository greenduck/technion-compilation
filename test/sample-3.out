(<PROGRAM,>(<FUNCTION,>(<FUNC_RET,>(<TYPE,>(<integer,>))
(<function,>))
(<FUNC_ARG,>(<(,>)
(<id,param5>)
(<:,>)
(<TYPE,>(<real,>))
(<,,>)
(<id,param2>)
(<:,>)
(<TYPE,>(<integer,>))
(<),>))
(<BLK,>(<DECLARATIONS,>(<var,>)
(<DECLARLIST,>(<DCL,>(<IDENTS,>(<id,b>))
(<:,>)
(<TYPE,>(<integer,>))
(<;,>))
(<DECLARLIST,>(<EPSILON,>))))
(<begin,>)
(<LIST,>(<LIST,>(<LIST,>(<LIST,>(<LIST,>(<LIST,>(<EPSILON,>))
(<STMT,>(<ASSN,>(<id,LeftAssn>)
(<assign,>)
(<EXP,>(<TERM,>(<TERM,>(<TERM,>(<FACTOR,>(<(,>)
(<EXP,>(<call,>)
(<(,>)
(<EXP,>(<TERM,>(<FACTOR,>(<num,0.44>))))
(<,,>)
(<EXP,>(<TERM,>(<FACTOR,>(<num,15>))))
(<),>))
(<),>)))
(<mulop,*>)
(<FACTOR,>(<(,>)
(<EXP,>(<EXP,>(<TERM,>(<FACTOR,>(<id,plusParam1>))))
(<addop,+>)
(<TERM,>(<FACTOR,>(<id,plusParam2>))))
(<),>)))
(<mulop,/>)
(<FACTOR,>(<(,>)
(<EXP,>(<EXP,>(<TERM,>(<FACTOR,>(<id,minus1>))))
(<addop,->)
(<TERM,>(<FACTOR,>(<id,minus2>))))
(<),>))))
(<;,>))))
(<STMT,>(<CNTRL,>(<if,>)
(<BEXP,>(<BEXP,>(<BEXP,>(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,a>))))
(<relop,==>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,b>))))))))
(<or,>)
(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,c>))))
(<relop,<>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,d>))))))))
(<or,>)
(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,e>))))
(<relop,<>>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,f>))))))))
(<then,>)
(<STMT,>(<ASSN,>(<id,x>)
(<assign,>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,y>))))
(<;,>))))))
(<STMT,>(<CNTRL,>(<for,>)
(<(,>)
(<STMT,>(<ASSN,>(<id,i>)
(<assign,>)
(<EXP,>(<TERM,>(<FACTOR,>(<num,0>))))
(<;,>)))
(<BEXP,>(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,i>))))
(<relop,<>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,size>))))))))
(<;,>)
(<STMT,>(<ASSN,>(<id,i>)
(<assign,>)
(<EXP,>(<EXP,>(<TERM,>(<FACTOR,>(<id,i>))))
(<addop,+>)
(<TERM,>(<FACTOR,>(<num,1>))))
(<;,>)))
(<),>)
(<STMT,>(<READ,>(<read,>)
(<(,>)
(<id,book>)
(<),>)
(<;,>))))))
(<STMT,>(<CNTRL,>(<while,>)
(<BEXP,>(<BEXP,>(<BTERM,>(<BFAC,>(<not,>)
(<BFAC,>(<BPFAC,>(<(,>)
(<BEXP,>(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,tierd>))))
(<relop,==>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,tierd>))))))))
(<),>))))))
(<or,>)
(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,ran>))))
(<relop,<>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,yoni>))))))))
(<do,>)
(<STMT,>(<RETURN,>(<return,>)
(<(,>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,coffie>))))
(<),>)
(<;,>))))))
(<STMT,>(<CNTRL,>(<if,>)
(<BEXP,>(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,first>))))
(<relop,==>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,exp>))))))))
(<then,>)
(<STMT,>(<CNTRL,>(<if,>)
(<BEXP,>(<BTERM,>(<BFAC,>(<BPFAC,>(<EXP,>(<TERM,>(<FACTOR,>(<id,second>))))
(<relop,==>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,exp2>))))))))
(<then,>)
(<STMT,>(<RETURN,>(<return,>)
(<(,>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,first>))))
(<),>)
(<;,>)))
(<else,>)
(<STMT,>(<WRITE,>(<write,>)
(<(,>)
(<EXP,>(<TERM,>(<FACTOR,>(<id,letter>))))
(<),>)
(<;,>))))))))
(<end,>)))
(<MAIN_FUNCTION,>(<program,>)
(<BLK,>(<DECLARATIONS,>(<var,>)
(<DECLARLIST,>(<DCL,>(<IDENTS,>(<id,a>))
(<:,>)
(<TYPE,>(<integer,>))
(<;,>))
(<DECLARLIST,>(<EPSILON,>))))
(<begin,>)
(<LIST,>(<EPSILON,>))
(<end,>))))
